src/DI/ContainerBuilder.php

```diff
@@ -755,6 +814,7 @@ public function getServiceName($arg)
 			}
 			return $res;
 		}
+		$service = isset($this->aliases[$service]) ? $this->aliases[$service] : $service;
 		if (!isset($this->definitions[$service])) {
```

Comment:
Imho `getServiceName()` should resolve aliases and return "canonical" service name. Call to `!hasDefinition($service)` is then redundant.